x-pack/legacy/plugins/lens/public/app_plugin/url_helper.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { DashboardConstants } from '../../../../../../src/legacy/core_plugins/kibana/public/dashboard';
+
+const EMPTY_DASHBOARD_PATTERN = /(.*#\/dashboard\?)(.*)/;
+const DASHBOARD_WITH_ID_PATTERN = /(.*#\/dashboard\/.*\?)(.*)/;
+
```

Comment:
This url_helper module seems like it should live in `src/legacy/core_plugins/kibana/public/dashboard` rather than in Lens. Lens really shouldn't know much about the internal details of dashboard URLs. The dashboard app should provide an API to adjust URLs. (This file could serve as that API.)

```js
import { addEmbeddable } from 'dashboard_url';

// Usage example...
const url = addEmbeddable(someDashboardUrl, 'lens', id);

/// Etc..
const url = addEmbeddable(someDashboardUrl, 'visualization', id);

// We may want to allow callers to modify other things?
const url = setDateRange(someDashboardUrl, dateRange;
```

Reply:
Good point. I wasn't sure where to put it, since Lens is the only thing that was using it.